mail_layout_force/models/mail_thread.py

```diff
@@ -18,3 +18,51 @@ def message_post_with_template(
         return super().message_post_with_template(
             template_id, email_layout_xmlid=email_layout_xmlid, **kwargs
         )
+
+    def _notify_record_by_email(
+        self,
+        message,
+        recipients_data,
+        msg_vals=False,
+        model_description=False,
+        mail_auto_delete=True,
+        check_existing=False,
+        force_send=True,
+        send_after_commit=True,
+        **kwargs
+    ):
+        msg_vals = msg_vals or {}
+        email_layout_xmlid = (
+            msg_vals.get("email_layout_xmlid")
+            if msg_vals
+            else message.email_layout_xmlid
+        )
+        template_xmlid = (
+            email_layout_xmlid
+            if email_layout_xmlid
+            else "mail.message_notification_email"
+        )
+        template = self.env.ref(template_xmlid, raise_if_not_found=True)
+        res_model = (
+            self.env["ir.model"].sudo().search([("model", "=", self._name)], limit=1)
+        )
+        layout_mappping = self.env["email.layout.mapping"].search(
+            [("layout_id", "=", template.id), ("models", "in", res_model.ids)], limit=1
+        )
+        if not layout_mappping:
+            layout_mappping = self.env["email.layout.mapping"].search(
+                [("layout_id", "=", template.id), ("models", "=", False)], limit=1
+            )
```

Comment:
```suggestion
        layout_mapping = self.env["email.layout.mapping"].search(
            [("layout_id", "=", template.id), "|", ("model_ids", "in", res_model.ids), ("model_ids", "=", False)], 
            limit=1,
        )
```

Reply:
I didn't combine them intentionally to avoid returning a record with `model_ids=False` even when a record with a matching `model_ids `exists.

Reply:
```
layout_mapping = self.env["email.layout.mapping"].search(
    [("layout_id", "=", template.id), "|", ("models", "in", res_model.ids), ("models", "=", False)],
    order="models desc",
    limit=1
)
```
@AungKoKoLin1997 How about this?

Reply:
It is not working. To be sure, I also tried `asc` but the same.